File: FXPRIMUS_JAVA/src/test/java/com/opcbiz/fxprimus/dto/qa19/AccountDto.java
package com.opcbiz.fxprimus.dto.qa19;

import java.util.ArrayList;
import java.util.List;

import com.opcbiz.fxprimus.dto.ExcelFieldDto;

public class AccountDto {
	String mt4;
	String id;
	String email;
	String passWord;
	String type;
	String refId;
	String accountInfor;

	public String getAccountInfor() {
		return accountInfor;
	}

	public void setAccountInfor(String accountInfor) {
		this.accountInfor = accountInfor;
	}

	public String getMt4() {
		return mt4;
	}

	public void setMt4(String mt4) {
		this.mt4 = mt4;
	}

	public AccountDto(String mt4, String id, String email, String passWord, String type, String accountInfor, String refId) {
		this.id = id;
		this.email = email;
		this.passWord = passWord;
		this.type = type;
		this.accountInfor = accountInfor;
		this.refId = refId;
		this.mt4 = mt4;
	}

	public static List<ExcelFieldDto> excelTemplate() {
		List<ExcelFieldDto> excelFieldDtoList = new ArrayList<ExcelFieldDto>();

		excelFieldDtoList.add(new ExcelFieldDto("MT4", "mt4", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("ID", "id", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("EMAIL", "email", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("PASSWORD", "passWord", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("TYPE", "type", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("REFERRAL_ID", "refId", ExcelFieldDto.DataType.STRING, "STRING"));
		excelFieldDtoList.add(new ExcelFieldDto("ACCOUNT_INFOR", "accountInfor", ExcelFieldDto.DataType.STRING, "STRING"));

		return excelFieldDtoList;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getPassWord() {
		return passWord;
	}

	public void setPassWord(String passWord) {
		this.passWord = passWord;
	}

	public String getID() {
		return id;
	}

	public void setID(String id) {
		this.id = id;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

	public String getRefId() {
		return refId;
	}

	public void setRefId(String refId) {
		this.refId = refId;
	}

	public String getWithIndex(int index) {
		switch (index) {
		case 1:
			return getMt4();
		case 2:
			return getID();
		case 3:
			return getEmail();
		case 4:
			return getPassWord();
		case 5:
			return getType();
		case 6:
			return getAccountInfor();
		case 7:
			return getRefId();

		}
		return "";

	}

	@Override
	public String toString() {
		return "AccountDto [email=" + email + ", passWord=" + passWord + ", ID=" + id + "]";
	}

}

File: src/opcbiz/fxprimus/dto/account_dto.py
from collections import namedtuple
from src.opcbiz.fxprimus.dto import excel_field_dto
from src.opcbiz.fxprimus.utils.excel_utils import ExcelUtils


class AccountDto:
    def __init__(self, id, email, pass_word,
                 account_type, account_status):
        self.id = id
        self.email = email
        self.pass_word = pass_word
        self.account_type = account_type
        self.account_status = account_status

    # getter method
    def get_id(self):
        return self.id

    def get_email(self):
        return self.email

    def get_pass_word(self):
        return self.pass_word

    def get_account_type(self):
        return self.account_type

    def get_account_status(self):
        return self.account_status

    # setter method
    def set_id(self, id):
        self.id = id

    def set_email(self, email):
        self.email = email

    def set_pass_word(self, pass_word):
        self.pass_word = pass_word

    def set_account_type(self, account_type):
        self.account_type = account_type

    def set_account_status(self, account_status):
        self.account_status = account_status

    def custom_dto_decoder(self):
        return namedtuple('AccountDto', self.keys())(*self.values())

    @staticmethod
    def excel_template():
        excel_field_dto_list = [excel_field_dto.ExcelFieldDto('ID', 'id'),
                                excel_field_dto.ExcelFieldDto('EMAIL', 'email'),
                                excel_field_dto.ExcelFieldDto('PASS_WORD', 'pass_word'),
                                excel_field_dto.ExcelFieldDto('ACCOUNT_TYPE', 'account_type'),
                                excel_field_dto.ExcelFieldDto('ACCOUNT_STATUS', 'account_status')]
        return excel_field_dto_list
